doc/user_mgmt/user_mgmt_hld.md

```diff
@@ -0,0 +1,595 @@
+
+
+![services_diagram](services_diagram.png))<br/>
+
+
+##  3 <a name='Configurationandmanagement'></a>Configuration and management
+
+###  3.1 <a name='ConfigDBTables'></a>ConfigDB Tables
+
+```
+USER_TABLE:{
+  "<username>":{
+    "state": {{enabled/disabled}}
+    "full-name": {{string}}
+    "hashed-password": {{string}}
+    "hashed-password_history": {{string}}
+    "role": {{admin/monitor}}
+    "passwd_date_of_last_change": {{uint32}}
+  }
+}
+```
+
+```
+ROLE_TABLE:{
+  "<role-name>":{
+    "primary_group": {{string}}
+    "secondary_groups": {{string}}
+  }
+}
+```
+###  3.2 <a name='ConfigDBschemas'></a>ConfigDB schemas
+
+```
+; Defines schema for User configuration attributes in USER_TABLE:
+key                                   = "<username>"                ;user configuration
+; field                               = value
+STATE                                 = "enabled" / "disabled"      ; user enable/disable
+FULL-NAME                             = STRING                      ; Full-name/Description of user
+HASHED-PASSWORD                       = STRING                      ; Hashed password
+HASHED-PASSWORD_HISTORY               = STRING                      ; List of old passwords (0-99)
+PASSWD_DATE_OF_LAST_CHANGE            = UINT32                      ; date of password last change
+ROLE                                  = "admin" / "monitor"         ; Role/Capability
+
+```
+
+```
+; Defines schema for role configuration attributes in ROLE_TABLE:
+key                                   = "<role name>"               ;Role configuration
+; field                               = value
+PRIMARY_GROUP                         = STRING                      ; Primary linux group
+SECONDARY_GROUP                       = STRING                      ; List of secondary linux groups
+```
+
+###  3.3 <a name='CLIYANGmodel'></a>CLI/YANG model
+
+##### YANG model
+```yang
+
+module sonic-user-mgmt {
+    yang-version 1.1;
+    namespace "http://github.com/Azure/sonic-user-mgmt";
+	prefix user-mgmt;
+
+    description "User Management YANG Module for SONiC OS";
+
+	revision 2022-12-01 {
+        description "First Revision";
+    }
+   container sonic-user-mgmt {
+
+        container USER_TABLE {
+
+            description "USER MANAGEMENT part of config_db.json";
+
+            list USER_TABLE_LIST {
+
+                key "name";
+
+                leaf name {
+                  type string {
+                        pattern '[a-zA-Z_]([a-zA-Z0-9_-]{0,31}|[a-zA-Z0-9_-]{0,30}$)';
+                        length 1..32 {
+                              error-message "Invalid length for the username.";
+                              error-app-tag username-invalid-length;
+                        }				  
+				          }
+                  description "User name.";
+                }                
+
+                leaf state {
```

Comment:
Generally state name is used for the read-only field, can we use "status" or some other appropriate name?

Reply:
it is already used in many other yang models
for example: [sonic-feature.yang](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-feature.yang)

Reply:
It's a suggestion to change the field name but I leave it upto you.